app/tests/test_search_utils.py

```diff
@@ -0,0 +1,311 @@
+                        "event_date": "30/10/2024",
+                        "creation_date": "20/08/2023",
+                    }
+                }
+            ],
+        ),
+        # no date field
+        (
+            [
+                {
+                    "_source": {
+                        "name": "Test Event",
+                        "description": "An event description",
+                    }
+                }
+            ],
+            [
+                {
+                    "_source": {
+                        "name": "Test Event",
+                        "description": "An event description",
+                    }
+                }
+            ],
+        ),
+        # empty date string
+        ([{"_source": {"event_date": ""}}], [{"_source": {"event_date": ""}}]),
+        # invalid date format (should return the original string)
+        (
+            [{"_source": {"event_date": "30-10-2024 13:45"}}],
+            [{"_source": {"event_date": "30-10-2024 13:45"}}],
+        ),
+        # empty results list
+        ([], []),
+    ],
+)
+def test_format_opensearch_results(results, expected):
+    actual = format_opensearch_results(results)
+    assert actual == expected
+
+
+@pytest.mark.parametrize(
+    "search_area, expected_fields",
+    [
+        # default "all" fields (no specific search area)
+        ("all", ["*"]),
+        # "metadata" search area
+        (
+            "metadata",
+            ["metadata_field_1", "metadata_field_2", "metadata_field_3"],
+        ),
+        # "record" search area
+        ("record", ["file_name", "file_path", "content"]),
+        # empty search_area string (should default to "all" behavior)
+        ("", ["*"]),
+        # none as search_area (should default to "all" behavior)
+        (None, ["*"]),
+        # invalid search_area (should default to "all" behavior)
+        ("invalid_area", ["*"]),
+    ],
+)
+@patch("app.main.util.search_utils.OpenSearch")
+@patch("app.main.util.search_utils.get_all_fields_excluding")
+def test_get_open_search_fields_to_search_on(
+    mock_get_all_fields_excluding, mock_opensearch, search_area, expected_fields
+):
+    mock_get_all_fields_excluding.return_value = [
+        "metadata_field_1",
+        "metadata_field_2",
+        "metadata_field_3",
+    ]
+    actual_fields = get_open_search_fields_to_search_on(
+        mock_opensearch, search_area
+    )
+    assert actual_fields == expected_fields
+
+
+def test_get_param(app):
+    with patch(
+        "flask.request.form.get", return_value="test_form_value"
+    ) as mock_form_get, patch(
+        "flask.request.args.get", return_value="test_arg_value"
+    ):
+
+        assert get_param("param") == "test_form_value"
+
+        mock_form_get.return_value = ""
+        assert get_param("param") == "test_arg_value"
+
+
+@patch(
+    "app.main.util.search_utils.get_param",
+    side_effect=["test_query", "test_area"],
+)
+def test_get_query_and_search_area(mock_get_param):
```

Comment:
we should not need to mock here, just make request a argument to `def get_param` and `get_query_and_search_area`